addon/components/ember-thead/component.js

```diff
@@ -195,7 +196,7 @@ export default Component.extend({
     this.addObserver('reorderFunction', this._updateApi);
 
     this.addObserver('sorts', this._updateColumnTree);
-    this.addObserver('columns', this._updateColumnTree);
+    this.addObserver('columns.[]', this._onColumnsChange);
```

Comment:
Without this, modification via mutation will not be noticed and `_onColumnsChange` is not called.

Reply:
Is it a viable alternative to require the invoking code to only provide new arrays for `columns`? More of an immutable pattern?

Reply:
We did discuss that @mixonic, but usage that we've seen really runs the gamut.  Feels a little awkward to disallow mutation with Ember compatible mutations.

Reply:
That would be one option, yes, and that would fix the problem. It wouldn't be hard to change it in our consuming app code (I don't think). But not making ember-table's awareness of `columns` kvo-compliant with Ember-provided kvo-compliant methods like `popObject` seems to me that it would violate the implicit contract that ember-table makes.